.github/workflows/ci.yml

```
@@ -0,0 +1,43 @@
+name: ci
+
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
```

Minor thing, these could be collapsed into:
```suggestion
on: [push, pull_request, workflow_dispatch]
```